internal/cmd/hana/map.go

```diff
@@ -183,28 +204,15 @@ func getClusterID(config *hanaCheckConfig) (string, error) {
 	return cm.Data["CLUSTER_ID"], nil
 }
 
-func isHanaReady(config *hanaCheckConfig) wait.ConditionWithContextFunc {
-	return func(_ context.Context) (bool, error) {
-		instance, err := getServiceInstance(config)
-		if err != nil {
-			return false, err
-		}
-
-		return isReady(instance)
-	}
-}
-
 func getHanaID(config *hanaCheckConfig) (string, error) {
-	instance := somethingWithStatus{}
-
 	// wait for until Hana instance is ready, for default setting it should take 5 minutes
 	fmt.Print("waiting for Hana instance to be ready... ")
-	err := wait.PollUntilContextTimeout(config.Ctx, 10*time.Second, config.timeout, true, isHanaReady(config))
+	err := wait.PollUntilContextTimeout(config.Ctx, 10*time.Second, config.timeout, true, kube.IsInstanceReady(config.kubeClient, config.Ctx, config.namespace, config.name))
```

Comment:
IMO this row is hard to read because of length. Could we calculate `kube.IsInstanceReady(...` befor this line?